### ISSUE TITLE
Text editor font changes not applying on Kubuntu 18.04

### ISSUE THREAD

**Issue by lyndhurst** [d1vanov/quentier#245]
Hi,
I have been trying out quentier on kubuntu 18.04, and I have not been able to change font size and colors so far.
I am using the [kvantum](https://github.com/tsujan/Kvantum) theme engine with a dark theme. So on fresh install, I have a black font on my dark themed editor background.
When I try and change font size and colors, the toolbar is responsive, and shows the changes I am making, but as soon as I set the focus back to the editor, all the toolbar settings are turned back to defaults instantly.
Trying to save the changes by clicking the 'save' toolbar button does not change the outcome. Other toolbar features like justify, code, bold, or italic do work fine.
If I can provide you more details, I will be glad to as I have been looking for a QT note taking application for quite a while now.

**Comment by d1vanov** [comment]
Hi,

it would be nice if you could provide some information about the used version of Quentier ("Help" menu -> "About Quentier") and where did you install it from. It would perhaps be also useful to provide [log](https://github.com/d1vanov/quentier/wiki/Collecting-Quentier-logs-for-troubleshooting) collected while reproducing the issue.

As a kind of a workaround you can try the following: select some word or character before changing the font size & color - it should be preserved then.

**Comment by lyndhurst** [comment]
Hi, I tried to emulate a few different cases, and record the logs. 
I will first list what I did in the correct sequence (try at least), then attach the log, I hope it will help you make sense of what is happening.

1. I typed some text with everything to defaults
2. I changed background color, saved, clicked back to editor 
==> no changes
3. I changed font size, saved, clicked back to editor 
==> no changes
4. I selected typed text, changed font color to white, as you suggested, saved, typed more text 
==> Selected text font color changed to white, changes did persist for additional text typed
5. I changed font size without selecting text, typed some text
==> Everything got reset to default
6. Selected text, changed font size and color, typed more
==> As you suggested settings did persist for additional text typed (font and color)

I think that my use of the app is not what was intended, and formatting should indeed happen after some text was selected. Anyway, I attach a screenshot with the version I am using, and another to show you the general default appearance of the text editor with my theme.

The software was conveniently installed through the stable ppa from the project page.

![Appearance](https://user-images.githubusercontent.com/30077629/55313792-f2e5f700-5468-11e9-9e80-807c21fd6f1d.png)
[quentier.log](https://github.com/d1vanov/quentier/files/3028391/quentier.log)
![Version](https://user-images.githubusercontent.com/30077629/55313794-f37e8d80-5468-11e9-9103-ad1541828af1.png)

**Comment by d1vanov** [comment]
Thank you for the detailed explanation, I'll look into it in details when I have some time for it.

I'd like to note a couple of things: first, I think your use is quite valid from a general user's point of view - you change the font size or color and expect it to stay - that's how the majority of apps with text editing work but Quentier doesn't. I kinda understand why it happens - the "current font" setting doesn't differentiate the cases when font size/color was selected and when it wasn't. Need to think how to fix it properly.

The second thing is, the note editor's background color should have stayed white - there was #221 about it. I'm going to try your theme and see why your background stays dark. In future I intend to implement more "proper" dark background mode.

**Comment by lyndhurst** [comment]
Perfect, let me know if I can help with further testing. By the way the theme I use is the mojave dark theme available by default in the kvantum theme engine linked above. Good luck.

**Comment by d1vanov** [comment]
I tried the mojave ark theme from kvantum theme engine and I'm kinda puzzled now: I confirm that Quentier version from launchpad has dark note editor background. But both my local build and AppImage have white note editor backgrounds. Something doesn't add up here, will investigate further.

**Comment by lyndhurst** [comment]
Ok, so, I am just understanding now that quentier is supposed to have a white editor background even with dark themes. 
I personally did not dislike at all that quentier did integrate well with the theme. More and more users like to read on screens with a dark background and light fonts like the 'zeal' or firefox reader dark mode which I find very easing on the eyes.
But if it is not consistent with the other builds you got installed, it is a double enigma now :)

If it helps with the logic, I also had to cancel the transparency that comes with the theme originally when 3d graphics are working on ubunutu. Because on first launch, the note editor was transparent; There is a blur effect in the kvantum manager settings to make transparent windows more useable, but still, it did not find it very confortable to read.

I know little about qt, but playing around with qt designer, I know that for example the central main widget, or the toolbox widget are transparent with that theme, while the tabbed widget is not. Maybe it can be a pointer for you.
Thank you for looking into it.

**Comment by d1vanov** [comment]
I sort of found out the reason for non-white note editor's background: build from launchpad used QtWebKit note editor backend while AppImage and my local build used QtWebEngine. And their behaviour for some reason differs with the solution enforcing white background that I had before. It seems I managed to find another way to make the background white even for QtWebKit based note editor, I've just merged it to master, in a few hours launchpad should rebuild libqt5quentier0 package and then the update would be available.

I totally get it that many people like dark themes. I want to support one for Quentier's note editor too but this support needs to be done properly - colors need to be sane e.g. dark background should imply light font color and vice versa. It's not a trivial thing to implement so I haven't done it yet but I intend to do it at some point. For now Quentier needs to be at the very least somewhat workable - together with Evernote integration it means the traditional white background and black font color by default.

I'll take a look at the font preference not being respected further on.

**Comment by d1vanov** [comment]
I haven't solved the original problem of this issue yet but I did another thing: I made note editor colors follow the chosen theme with the possibility for the user to override these colors if necessary (through preferences dialog - "Service" -> "Settings" -> "Note editor" tab in stable version, or "Edit" -> "Preferences" -> "Note editor" tab in unstable version). Usually I only introduce new functionality only into the unstable version but I figured this is important enough to be present in stable version too.

Regarding the original problem of this issue - I figured why it happens but haven't found the solution yet, will keep looking.

**Comment by lyndhurst** [comment]
As you know, I have the stable ppa installed, and, just after our last exchange here, I had noticed a couple of updates, and I saw that you had the editor background back to white; I was a bit disappointed for the theme integration part, but glad it was gonna be useable/readable anyway. 
I just updated my repositories, and I can confirm that it is both useable, and "looking good" )
Meaning, theme integration is just as expected, and starting a new note, without changing any settings, I have a perfectly readable white font on a dark editor background.
Glad you could figure out what the problem was, and thank you again for the time and effort.

**Comment by d1vanov** [comment]
I did some more work here and what I have accomplished is this:

1. when you change font family and/or size after you've just created a note but before you typed anything into it, the font change would actually be respected for text entered further.
2. fixed a bug which caused wrong detection of font when you set cursor to some arbitrary place within the note.

Both changes were done in development i.e. unstable branch only for now. I intend to finalize a couple more things in development, do some proper testing and merge the development to master i.e. make it the new stable version. Can't say when it would be done but I hope within a couple of months.

I'd also like to extend the behaviour on changing font after some content has been entered into a note too but I'll work on that in context of #246. I did some research and seemingly found a way to do that but it would require quite some work.
